bin/deploy

```diff
@@ -52,6 +52,7 @@ done
 
 echo "Deploying $NETWORK_NAME network...";
 
+# TODO: this fails without an error if not at the HEAD of a given branch
```

Comment:
@strophy How does this fail? From my testing it will set BRANCH to an empty string, but that looks to be a valid condition below, since an empty string won't match anything.

Reply:
@nmarley it fails with exit code `1`. Try this to reproduce:
```
git checkout v0.23-dev
git symbolic-ref --short -q HEAD // returns v0.23-dev
git checkout 7d32585bf04bd9510367064038d0a7ac31b02765
git symbolic-ref --short -q HEAD // fails
```

Reply:
Ok, now I understand. The script fails because `-e` is set above to fail upon any error. If you want an error you can just remove the `-q`:
```
-q, --quiet
    Do not issue an error message if the <name> is not a symbolic ref but a detached HEAD;
    instead exit with non-zero status silently.

Reply:
Ok, now I understand. The script fails because `-e` is set above to fail upon any error. If you want an error you can just remove the `-q` from the `git symbolic-ref` command:

```
-q, --quiet
    Do not issue an error message if the <name> is not a symbolic ref but a detached HEAD;
    instead exit with non-zero status silently.
```

Or we can discuss a different fix internally, but that should be better than silently failing I would think.

Reply:
@nmarley just an error is fine for now: https://github.com/dashpay/dash-network-deploy/pull/393